Tasks/Sherel/Task7/Task7_5/Task7_5(json+ram+decorator)/Task7(combo).py

```diff
@@ -0,0 +1,44 @@
+"""
+How the program works: I open a json file - which stores the history of my calculations in the long run
+(if the file is missing, I create an empty one) and take the entire history from it into my temporary dictionary,
+which will act as my short-term cache, since reading file consumes more PC resources than its one-time iteration.
+If my new calculation is not in my dictionary stored in memory, I add to ram and add cache to json,
+while all subsequent calculations are checked only by ram memory so as not to constantly open json for reading,
+but only write to it.
+"""
+import json
+
+file_path = 'cache.json'
+try:
+    with open(file_path, 'r') as json_file:
+        data = json.load(json_file)
```

Comment:
basically here you have mixed cache for all the functions which would be decorated, it's not exactly the best approach in every possible case